sympy/solvers/solveset.py

```diff
@@ -3665,33 +3668,32 @@ def nonlinsolve(system, *symbols):
         return _solveset_work(system, symbols)
 
     # main code of def nonlinsolve() starts from here
-    polys, polys_expr, nonpolys, denominators = _separate_poly_nonpoly(
-        system, symbols)
 
-    if len(symbols) == len(polys):
-        # If all the equations in the system are poly
+    # only consider equations with either no free symbols
+    # or at least free symbol in common with symbols to solve for
```

Comment:
If there is no free symbol in an equation, what is the purpose of including it in the equations to be processed?

Reply:
If there is no symbol of interest then a non-zero expression generates and EmptySet while a potentially zero expression gives a solution.
```python
>>> nonlinsolve((sqrt(3),x**2-4),(x,))
EmptySet
>>> nonlinsolve((sqrt(y),x**2-4),(x,))
{(-2,), (2,)}
```
Would it be better to return some sort of set that requires that `Ne(sqrt(y), 0)`?

Reply:
> If there is no symbol of interest then a non-zero expression generates an EmptySet while a potentially zero expression allows a solution (but makes no reference to that condition).

I wrote it like that originally because `solve` does the same thing. 
```python
>>> solve((sqrt(3),x**2-4), x)
[]
>>> solve((sqrt(y),x**2-4), x)
[(-2,), (2,)]
```
> Would it be better to return some sort of set that requires that Ne(sqrt(y), 0)?

I'm not sure what you mean here. Do you mean requiring that `Eq(sqrt(y), 0)`?. If so, does sympy have a set object that evaluates to one set (`{(-2,), (2,)}` in this case) if a condition (`Eq(sqrt(y), 0)`) is true and another set (`EmptySet`) otherwise? Even if it does, I don't know how useful such behavior will be to users, despite it being mathematically correct. What do you think?

Reply:
I think that's a bug in `solve`.

Reply:
So what should be the return value of `nonlinsolve((sqrt(y),x**2-4),(x,))` then? Should it be `EmptySet`?

Reply:
Yes it should be empty set. The situation is similar to this:
```python
In [4]: solve([x, x+y], [x])
Out[4]: []
```
Here there is a solution `x=0` but only if `y=0`. Here `y` is a parameter of the system of equations rather than one of the unknowns to be solved for so we can't just choose for it to have the value `0`: for almost all possible values of `y` the system of equations does not have any solution for `x` that can make all equations zero.